src/libslic3r/GCode/GCodeProcessor.cpp

```diff
@@ -668,7 +668,10 @@ void GCodeProcessor::TimeProcessor::post_process(const std::string& filename, st
                 if (eol) {
                     ++line_id;
 
-                    gcode_line += "\n";
+                    // determine the end of line character and pass to output
```

Comment:
Do you mind to expain a little bit on this change?

Reply:
Yeah. The biqu firmware is sensitive to the end of line character used. If you use only \n, it doesn't work, only \r\n. The original code in the post processor always replaced any EOL character with \n. This change gets the detected EOL character and if it is \r, it checks if the next character. If the next character is \n, it adds the \n (to complete the \r\n). Basically, it preserves the EOL character(s) from the pre post processed gcode.